frontend/src/__tests__/components/flex.test.js

```diff
@@ -0,0 +1,63 @@
+import '@testing-library/jest-dom';
+import { render, screen } from '@testing-library/react';
+import Flex from '../../../components/Flex/Flex';
+import Colors from '../../../components/colors';
+
+describe('Flex Component Tests', () => {
+  it('should render a default Flex component', () => {
+    render(<Flex />);
+    const flexComponent = screen.getByTestId('flex-1');
+    expect(flexComponent).toBeInTheDocument();
+    expect(flexComponent).toHaveStyle(
+      `background-color : ${Colors.Mono[0]}`,
+      'gap:unset',
+      'flex-wrap:wrap',
+      'margin:0',
+      'width:100%',
+      'height:100%',
+      'padding:0',
+      'justify-content:center',
+      'align-content:center',
+      'flex-direction:initial',
+      'z-index:0',
+      'transform:inherit',
+      'border-radius:0'
+    );
+  });
+  it('should render a Flex component with specified unique style', () => {
+    render(
+      <Flex
+        bgColor={Colors.Mono[1]}
+        gap={'10px'}
+        wrap={'nowrap'}
+        margin={'10'}
+        width={'50%'}
+        height={'50%'}
+        padding={'10'}
+        justifyContent={'flex-start'}
+        alignContent={'flex-start'}
+        direction={'column'}
+        zIndex={'1'}
+        transform={'translateY(-10px)'}
+        borderRadius={'15px'}
+      />
+    );
+    const flexComponent = screen.getByTestId('flex-1');
```

Comment:
Also, I would implore you to try and use something other than `testId`'s. This method of getting elements from the DOM should be used as a last case scenario, as there are more specific selectors that RTL offers to help you select your components. I would also go as far as to say, if you have to use `getByTestId` you probably did not format your HTML semantics correctly and in a way that makes sense.

Reply:
I am making adjustments to the components that make sense to use alternate means. I am however a bit stuck with the Flex component, as it is just a div, and not something that has a role. For those types of components do i need to just make an aria-label that i can then search by, or am I missing something?

Reply:
I wonder if we can do something with a div and use one of RTL's other selectors so that we don't pollute our code with testId's?

Remember, you're testing a component. If it's a div that is meant to hold content (i.e. children), you need to test that. Perhaps that will help you find your way to the correct selector.